### PR TITLE
Consolidate Incident Endpoints

### PR DESCRIPTION
**Pull request by abe-one**
# Consolidates all incident endpoints into two DRY endpoints
Fixes Inconsistent Incident Endpoints
https://trello.com/c/cKJ2LqWP/34-consolidate-fragmented-endpoints

## Type of change
Please delete options that are not relevant.
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Change Status
- [x] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

# Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] There are no merge conflicts

### CODE REVIEW

File: api/allIncidents/adminIncidentsRouter.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+ *    produces:
+ *      - application/json
+ *    responses:
+ *      201:
+ *        description: Success ... returns updated incident object
+ *      500:
+ *        description: Server response error
+ */
+
+router.put(
```
Comment (francishtknguyen) [inline api/allIncidents/adminIncidentsRouter.js:83]: is there a way to get this formatted like the other routes? just nitpicking

Comment (abe-one) [reply api/allIncidents/adminIncidentsRouter.js:83]: Formatted in what sense? Using Try Catch?

Comment (francishtknguyen) [reply api/allIncidents/adminIncidentsRouter.js:83]: just have them inline instead of cascading, but I think that has to do with the formatter

Comment (abe-one) [reply api/allIncidents/adminIncidentsRouter.js:83]: You mean the then/catch?

File: api/allIncidents/incidentsRouter.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ *      - application/json
+ *    responses:
+ *      200:
+ *        description: Success ... returns an array of incident objects
+ *      500:
+ *        description: Server response error
+ */
+
+router.get('/', (req, res, next) => {
+  Incidents.getApprovedIncidents()
```
Comment (francishtknguyen) [inline api/allIncidents/incidentsRouter.js]: Will this really be getting only approved incident's or would that be something filtered in the front end after pulling all of the incidents?

Comment (abe-one) [reply api/allIncidents/incidentsRouter.js]: That will be your model function querying the database for only approved incidents :)

File: api/allIncidents/incidentsRouter.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    next({ status: 400, message: 'Incident unavailable' });
+  }
+});
+
+// eslint-disable-next-line no-unused-vars
+router.use((err, _req, res, _next) => {
+  res
+    .status(err.status || 500)
+    .json({ message: err.message || 'Database Error' });
+});
```
Comment (JustinPeczenij) [inline api/allIncidents/incidentsRouter.js:66]: Public endpoint to get the 5 most recent incidents to populate the timeline?

File: api/util/validationSchemas.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  city: yup
+    .string()
+    .trim()
+    .min(2, 'city must be at least 2 characters long')
+    .max(50, 'city must be at most 50 characters long'),
+
+  startDate: yup.string().trim(),
+
+  endDate: yup.string().trim(),
+});
```
Comment (JustinPeczenij) [inline api/util/validationSchemas.js:19]: Good idea to validate on the backend

**Review by JustinPeczenij — DISMISSED** [review]

File: api/allIncidents/incidentsRouter.js (hunk 3 above)
Comment (abe-one) [reply api/allIncidents/incidentsRouter.js:66]: AH, you're right!

**Review by JustinPeczenij — APPROVED** [review]
Looks good.

File: api/util/validationSchemas.js (hunk 4 above)
Comment (abe-one) [reply api/util/validationSchemas.js:19]: Yeah, we'll need a future PR to properly sanitize the data. Posted a comment on the Trello

**Comment by francishtknguyen** [comment]
Looks good !

**Review by francishtknguyen — APPROVED** [review]

**Review by JustinPeczenij — APPROVED** [review]

**Pull request merged**